rocketpy/mathutils/function.py

```diff
@@ -2427,6 +2427,30 @@ def differentiate(self, x, dx=1e-6, order=1):
                 + self.get_value_opt(x - dx)
             ) / dx**2
 
+    def differentiate_complex_step(self, x, dx=1e-6):
+        """Differentiate a Function object at a given point using the complex
+        step method. This method can be faster than ``Function.differentiate``
```

Comment:
Shouldn't `dx` be much smaller for this to be efficient?

Reply:
Idk I didn't have much time to investigate it.

Copilot suggested 1e-20 but I changed to 1e-6 to make it more similar to the differentiate method.

All in all it is an user choice, I just set the standard value.
From the tests I implemented, we could see that it has to be sufficient. 

Would you still reduce the dx default value even more?

Reply:
Now I see the recommendation to use 1e-200, seems cool.

Reply:
Solved here: [1356248](https://github.com/RocketPy-Team/RocketPy/pull/594/commits/13562483f646d8cb61f2105d1fa72f892aa0a37c)